### PR TITLE
Batch messages to terminal before terminal initialized

### PR DESCRIPTION
**Pull request by christianalfoni**

### CODE REVIEW

File: packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx [hunk 1]
```diff
@@ -34,6 +32,7 @@ class DevToolTerminal extends React.Component<
   };
 
   term: TerminalWithFit;
+  batchedMessages: any[];
```
Comment (lbogdan) [inline packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx]: I'd name this `messageQueue` or `queuedMessages`?

**Comment by lbogdan** [comment]
Fixes

image

File: packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx (hunk 1 above)
Comment (christianalfoni) [inline packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx]:
```suggestion
  messageQueue: any[];
```

File: packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx [hunk 2]
```diff
@@ -46,9 +45,17 @@ class DevToolTerminal extends React.Component<
 
   setTerminal = (terminal: TerminalWithFit) => {
     this.term = terminal;
+
+    this.batchedMessages.forEach(this.handleMessage);
```
Comment (christianalfoni) [inline packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx]:
```suggestion
    this.messageQueue.forEach(this.handleMessage);
```

File: packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx [hunk 3]
```diff
@@ -46,9 +45,17 @@ class DevToolTerminal extends React.Component<
...
+    this.batchedMessages.length = 0;
```
Comment (christianalfoni) [inline packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx]:
```suggestion
    this.messageQueue.length = 0;
```

File: packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx [hunk 4]
```diff
@@ -46,9 +45,17 @@ class DevToolTerminal extends React.Component<
...
   };
 
   handleMessage = (data: any) => {
+    if (!this.term) {
+      this.batchedMessages.push(data);
```
Comment (christianalfoni) [inline packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx]:
```suggestion
      this.messageQueue.push(data);
```

File: packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx (hunk 1 above)
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/Terminal/index.tsx]: Ah... fixed, by suggesting code changes in Github directly... soooo, lazy 😂

**Pull request merged**